doc/reference/trd-legal.md

```diff
@@ -0,0 +1,183 @@
+Licensing and Copyrights
+========================================
+
+**TRD:** <br/>
+**Working Group:** Core<br/>
+**Type:** Best Current Practice<br/>
+**Status:** Draft <br/>
+**Author:** Pat Pannuto<br/>
+**Draft-Created:** 2022/11/05 <br/>
+**Draft-Modified:** 2022/11/05 <br/>
+**Draft-Version:** 1 <br/>
+**Draft-Discuss:** tock-dev@googlegroups.com<br/>
+
+Abstract
+--------
+
+This document describes Tock’s policy on licensing and copyright. It explains
+the rationale behind the license selection (dual-license, MIT or Apache2) and
+copyright policy (optional, and authors may retain copyright if desired). It
+further outlines how licensing and copyright shall be handled throughout
+projects under the Tock umbrella.
+
+Explicitly not discussed in this TRD are issues of trademark, the Tock brand,
+logos, or other non-code assets and artifacts.
+
+
+1. Introduction
+===============
+
+Tock’s goal is to provide a safe, secure, and efficient operating system for
+microcontroller-class devices. The Tock project further believes it is
+important to enable and support widespread adoption of safe, secure, and
+efficient software. Tock also seeks to be an open and inclusive project, and
+Tock welcomes contributions from any individuals or entities wishing to
+improve the safety, security, reliability, efficiency, or usability of Tock
+and the Tock ecosystem.
+
+The intent of these policies is to best satisfy the needs of all stakeholders
+in the Tock ecosystem.
+
+
+2. Licensing
+============
+
+All software artifacts under the umbrella of the Tock project are
+dual-licensed as Apache License, Version 2.0 (LICENSE-APACHE or
+http://www.apache.org/licenses/LICENSE-2.0) or MIT license (LICENSE-MIT or
+http://opensource.org/licenses/MIT).
+
+All contributions to Tock projects MUST be licensed under these terms.
+
+
+3. Copyright
+============
+
+Entities contributing resources to open-source projects often require
+attribution to recognize their efforts. Copyright notices are a common means to
+provide this. For downstream users, the license terms of Tock ensures
+unencumbered use.
+
+Copyrights in Tock projects are retained by their contributors. No
+copyright assignment is required to contribute to Tock projects.
+
+Artifacts in the Tock project MAY include explicit copyright notices.
+Substantial updates to an artifact MAY add additional copyright notices
+to an artifact. Copyright notices SHOULD NOT be removed by any party except
+the original copyright holder.
+
+For full authorship information, see the version control history.
+
+
+4. Implementation
+=================
+
+All code artifacts in Tock projects MUST include a license notice.
```

Comment:
What about files that do not support comments? Markdown comes to mind -- I suspect we want to omit license notices in our Markdown documents.

Reply:
I suspect that is fine.

Reply:
It sounds like you two have different answers to the question "is markdown code?"

Reply:
I'm not sure that either "Markdown vs not-Markdown" or "code vs not-code" is the right place to make the distinction. How about requiring headers in all file formats that support comments? I don't want someone to think that images have to have license info embedded in their EXIF tags, so maybe all *textual* file formats that support comments.

I don't particularly care where we draw the line, I just think we need to be unambiguous.

Reply:
Sure, that seems fine to me.

Reply:
Conclusion: Textual files that allow comments

(e.g. Markdown files don't support comments, so we don't expect it)